### ISSUE TITLE
Ability to use 1Password Document items

### ISSUE THREAD

**Issue by mattparkes** [1Password/onepassword-operator#47]
### Summary
We want to be able to use secrets in 1Password which are large multi-line strings. Examples include PEM files, or even the `scimsession` file for `op-scim-bridge`.

### Use cases
It's extremely common for enterprise secrets to be multiline strings. The only way to store these in 1Password is using the Document type. (As an aside, it would be preferable if you could add Document type fields to any Password Item, but that is outside the scope of this ticket)

However, the issue is that if you create one of these items in 1Password and then reference it with a `OnePasswordItem`, you get a `Secret` with no `data`.

### Proposed solution
If the Item is a Document, then the file should be in the `data` section of the Secret as:
`<filename>: <file contents>`

### Is there a workaround to accomplish this today?
None. I tried adding a literal`\n` character but unsurprisingly, it didn't work.

### References & Prior Work
 - Kubernetes secrets can be multiline: https://kubernetes.io/docs/concepts/configuration/secret/ (they are just base64 encoded strings, so they decode to multiline strings as expected)
 - Hashicorp Vault handles multiline strings: https://vault-cli.readthedocs.io/en/latest/howto/write.html
 - etc

**Comment by camilb** [comment]
I've tried to achieve this as well and found that if you set the `itemPath` to a 1Password item that has the documents linked, it will create the secret with the correct keys, but the keys values are completely wrong. I was trying to create some certificates   (certificate/private-key) and you can see the results in the screenshot.

![ScreenShot2021-06-25 at 12 52 26@2x](https://user-images.githubusercontent.com/2655869/123416763-44817b80-d5b7-11eb-93a0-f02f2c8d85c9.png)
![ScreenShot2021-06-25 at 12 53 41@2x](https://user-images.githubusercontent.com/2655869/123416759-43504e80-d5b7-11eb-93e6-6e28ec48c164.png)

**Comment by mdaniel** [comment]
Is this a specialization of #38 or does it need to be its own issue?

**Comment by mattparkes** [comment]
It's entirely different/opposite and needs to be its own issue.

#38 is about making items in 1Password vaults go into different [Kubernetes Secret Types](https://kubernetes.io/docs/concepts/configuration/secret/#secret-types). This ticket is about making different 1Password types (specifically documents/files) go into Kubernetes Secrets (presumably just the 1 type since that's all that is supported until #38 is addressed).

**Comment by jillianwilson** [comment]
Hi there,

Thanks for bringing this to our attention! I think supporting documents in the Kubernetes Operator is definitely a useful feature that we hope to add in the future. We will keep you updated on any progress on that front.

**Comment by davidgibbons** [comment]
Either documents or secure notes would really be useful. Secure notes _almost_ work as needed but since the key name gets set to notesPlain it's basically useless.  Being able to relabel/rekey it in the CRD would be a great mitigation so we can actually write out a multi-line file.

For example: 
` k get secrets test -o yaml
apiVersion: v1
data:
  notesPlain: W2RlZmF1bHRdCmF3c19hY2Nlc3Nfa2V5X2lkID0gdGhpc2lzYXRlc3QKYXdzX3NlY3JldF9hY2Nlc3Nfa2V5ID0gdGVzdHNlY3JldA==
kind: Secret`

`$ echo W2RlZmF1bHRdCmF3c19hY2Nlc3Nfa2V5X2lkID0gdGhpc2lzYXRlc3QKYXdzX3NlY3JldF9hY2Nlc3Nfa2V5ID0gdGVzdHNlY3JldA== |base64 -d
[default]
aws_access_key_id = thisisatest
aws_secret_access_key = testsecret`

I can actually get a multi-line secret to work this way, but since I can't control the key name (notesPlain) it's situational and difficult to use.

**Comment by bygui86** [comment]
@jillianwilson any news about this feature?
Is it in the roadmap?
Does someone work on it already?
Do you have a rough release date?

thank you :)

**Comment by tofarley** [comment]
Also watching this closely. The ability to store SSH keys and TLS certs is an absolute must-have before we can complete onboarding of 1Password.

k8s often expects the data field

```
data:
  ca.crt: multi-line string
  tls.crt: multi-line string
  tls.key: multi-line string
```

The ability to replicate this secret with multiline fields is imperative.

**Comment by mattparkes** [comment]
@jillianwilson is it possible to get any sort of ETA on this? #82 has also been opened recently and all the reactions in this thread indicate that this is definitely something people are keen for.

**Comment by oyamin** [comment]
@jillianwilson +1 for this feature. Is there any possible updates regarding this? Our company is integrating 1password with our k8s cluster, and this feature is currently blocking us from from fully migrating to 1password.

**Comment by bygui86** [comment]
@oyamin +1 here! Our company evaluates 1password operator but without this feature is really limited in usage.
Unfortunately I think that 1password company is under reorganisation.

No commits since Nov 2021... PR opened since ages... I have even the impression that this project could have been abandoned...

@jillianwilson can you please answer somewhere?

**Comment by oyamin** [comment]
@bygui86 There is actually a PR already submitted back in November, that almost does what we need. https://github.com/1Password/onepassword-operator/pull/80

**Comment by bygui86** [comment]
@oyamin that's great but is blocked... it would be great to know when could be merged, but nobody from 1Password answers...

**Comment by mattparkes** [comment]
I had a call with a 1 Password CSM this morning and raised this issue with them and I stressed the importance of this feature for most people's workflows with this Operator.

They couldn't make any promises on a specific ETA because of other in-progress work, but they do have eyes on this ticket.

I also discussed with them that it's frustrating that this project hasn't had any love for a little while, which they fully agree with (and to their credit, they have started pulling in some PRs over the past few days).

**Comment by littlefixit** [comment]
Was this resolved in #87 ? If So, I'm having difficulty figuring it out.

**Comment by Marton6** [comment]
Support for Document items, and, more broadly, for loading data from files stored in 1Password into Kubernetes secrets has just been added in [`v1.3.0`](https://github.com/1Password/onepassword-operator/releases/tag/v1.3.0).

**Comment by oyamin-absci** [comment]
Can you give us an example on how to use this new feature? Since I tried setting my items in document and it just pulls an empty secret:
```
NAME                        TYPE                                  DATA
env-file                    Opaque                                0      
```
Using:
```
apiVersion: onepassword.com/v1
kind: OnePasswordItem
metadata:
  name: env-file
spec:
  itemPath: "vaults/dev/items/env_file"
```
And the file just had couple entries for ENV vars
var_1=test123
var_2=test456

**Comment by oyamin** [comment]
Can you give us an example on how to use this new feature? Since I tried setting my items in document and it just pulls an empty secret:
```
NAME                        TYPE                                  DATA
env-file                    Opaque                                0      
```
Using:
```
apiVersion: onepassword.com/v1
kind: OnePasswordItem
metadata:
  name: env-file
spec:
  itemPath: "vaults/dev/items/env_file"
```
And the file just had couple entries for ENV vars
var_1=test123
var_2=test456

**Comment by Varjokorento** [comment]
@Marton6 is there an example on how to use this feature? I tried the same way as I do for all the other secrets and ended up with empty data (same thing that happened with @oyamin )
